src/api/utils/gasOracle.ts

```diff
@@ -75,6 +79,12 @@ export class GasOracleService {
     }
   }
 
+  private async getOptimismGasPrice(): Promise<number> {
+    const price =  await this.provider.getGasPrice();
+
+    return price.toNumber();
+  }
```

Comment:
It turns out there's a url endpoint to get the optimism gas price via their etherscan site here. 

https://api-optimistic.etherscan.io/api?module=proxy&action=eth_gasPrice

If we used that we wouldn't need to pass in the provider / change the instantiation flow here.

Reply:
Thanks for sharing! wasn't aware of this. I'll use that instead.

Reply:
one comment is that adding an api endpoint means we have to update our CSP allow list on our UI